include/circt/Dialect/FIRRTL/InstanceGraph.h

```diff
@@ -182,6 +182,38 @@ class InstanceGraph {
   llvm::StringMap<unsigned> nodeMap;
 };
 
+/// An absolute instance path.
+using InstancePath = ArrayRef<InstanceOp>;
+
+template <typename T>
+static T &operator<<(T &os, const InstancePath &path) {
+  os << "$root";
+  for (auto inst : path)
+    os << "/" << inst.name() << ":" << inst.moduleName();
+  return os;
+}
+
+/// A data structure that caches and provides absolute paths to module instances
+/// in the IR.
+struct InstancePaths {
```

Comment:
InstancePath and InstancePaths?  Maybe InstancePathCache?

Reply:
Changing this to `InstancePathCache` in the rebase.